File: 2728/Grace Hopper a Vovó do Cobol.cpp
#include <iostream>
#include <vector>
#include <bits/stdc++.h>
using namespace std;


int main()
{
    string word, input;
    vector<string> list;
    while (cin >> input)
    {




        bool cobol = false;
        stringstream sentence(input);

        while (getline(sentence, word, '-'))
        {
            list.push_back(word);
        }


        if (tolower(list[0][0]) == 'c' || tolower(list[0][list[0].size() - 1]) == 'c')
        {
            if (tolower(list[1][0]) == 'o' || tolower(list[1][list[1].size() - 1]) == 'o')
            {
                if (tolower(list[2][0]) == 'b' || tolower(list[2][list[2].size() - 1]) == 'b')
                {
                    if (tolower(list[3][0]) == 'o' || tolower(list[3][list[3].size() - 1]) == 'o')
                    {
                        if (tolower(list[4][0]) == 'l' || tolower(list[4][list[4].size() - 1]) == 'l')
                        {
                            cobol = true;
                        }
                    }
                }
            }

        }

        if (cobol)
        {
            cout << "GRACE HOPPER" << endl;

        }
        else
        {
            cout << "BUG" << endl;
        }
        list.clear();
    }
}

File: 2728/Grace Hopper a Vovó do Cobol.py

while True:
    try:
        words = input().split("-")
        cobol = False

        if words[0][0].lower() == 'c' or words[0][len(words[0])-1].lower() == 'c':
            if words[1][0].lower() == 'o' or words[1][len(words[1])-1].lower() == 'o':
                if words[2][0].lower() == 'b' or words[2][len(words[2])-1].lower() == 'b':
                    if words[3][0].lower() == 'o' or words[3][len(words[3])-1].lower() == 'o':
                        if words[4][0].lower() == 'l' or words[4][len(words[4])-1].lower() == 'l':
                            cobol = True


        if cobol:
            print("GRACE HOPPER")
        else:
            print("BUG")
    
    except EOFError:
        break
